app/src/main/java/com/example/georgesamuel/dubaihotels/RxJava/RxJavaOperatorsActivity.java

```diff
@@ -0,0 +1,199 @@
+package com.example.georgesamuel.dubaihotels.RxJava;
+
+import android.os.Bundle;
+import android.util.Log;
+
+import androidx.annotation.Nullable;
+import androidx.appcompat.app.AppCompatActivity;
+
+import com.example.georgesamuel.dubaihotels.R;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Random;
+
+import io.reactivex.Observable;
+import io.reactivex.ObservableEmitter;
+import io.reactivex.ObservableOnSubscribe;
+import io.reactivex.Observer;
+import io.reactivex.android.schedulers.AndroidSchedulers;
+import io.reactivex.disposables.Disposable;
+import io.reactivex.functions.Function;
+import io.reactivex.schedulers.Schedulers;
+
+public class RxJavaOperatorsActivity extends AppCompatActivity {
+
+    private static final String TAG = RxJavaOperatorsActivity.class.getSimpleName();
+    private Disposable mapDisposal;
+    private Disposable flatMapDisposable;
+
+    @Override
+    protected void onCreate(@Nullable Bundle savedInstanceState) {
+        super.onCreate(savedInstanceState);
+        setContentView(R.layout.activity_rx_java);
+
+        //useMapOperator();
+        //useFlatMapOperator();
+    }
+
+    private void useFlatMapOperator(){
+        getUserObservable().subscribeOn(Schedulers.io())
+                .observeOn(AndroidSchedulers.mainThread())
+                .flatMap(new Function<User, Observable<User>>() {
+                    @Override
+                    public Observable<User> apply(User user) throws Exception {
+                        return getEmailObservable(user);
+                    }
+                })
+                .subscribe(new Observer<User>() {
+                    @Override
+                    public void onSubscribe(Disposable d) {
+                        Log.d(TAG, "onSubscribe");
+                        flatMapDisposable = d;
+                    }
+
+                    @Override
+                    public void onNext(User user) {
+                        Log.d(TAG, "onNext: " + user.getName() + " " + user.getEmail());
+                    }
+
+                    @Override
+                    public void onError(Throwable e) {
+                        Log.d(TAG, "onError: " + e.getMessage());
+                    }
+
+                    @Override
+                    public void onComplete() {
+                        Log.d(TAG, "onComplete: FlatMap");
+                    }
+                });
+    }
+
+    private Observable<User> getEmailObservable(final User user){
```

Comment:
this code didn't trigger onNext , instead i add something like that and it was working
 private void useFlatMapOperator() {
        getUserObservable().subscribeOn(Schedulers.io())
                .observeOn(AndroidSchedulers.mainThread())
                .flatMap(new Function<User, Observable<User>>() {
                    @Override
                    public Observable<User> apply(User user) throws Exception {
                        Email email = new Email();
                        email.setEmail("any");
                        user.setEmail(email.getEmail());
                        return Observable.just(user);
                    }
                })
                .subscribe(new Observer<User>() {
                    @Override
                    public void onSubscribe(Disposable d) {
                        Log.d(TAG, "onSubscribe");
                        flatMapDisposable = d;
                    }

                    @Override
                    public void onNext(User user) {
                        Log.d(TAG, "onNext: " + user.getName() + " " + user.getEmail());
                    }

                    @Override
                    public void onError(Throwable e) {
                        Log.d(TAG, "onError: " + e.getMessage());
                    }

                    @Override
                    public void onComplete() {
                        Log.d(TAG, "onComplete: FlatMap");
                    }
                });